services/gossip/codec.go

```diff
@@ -0,0 +1,82 @@
+package gossip
+
+import (
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+)
+
+func encodeBlockPair(blockPair *protocol.BlockPairContainer) ([][]byte, error) {
```

Comment:
hmm.. not sure it's a huge chunk of memory because it's only pointers. Let's assume you have a block with 1000 transactions and 1000 transaction receipts. Each transaction can be 1KB so the block itself will be over 1MB in size. But the memory chunk here will only be 2000*pointersize = 8KB-16KB which isn't too bad..

I do agree that we will eventually need to move to a Writer instead of []byte.. but I'll leave that to the gossip team :)

Reply:
I don't understand your math. If the outcome of this method is a fully-serialized block-pair to be transmitted over TCP, it will be 1MB. What am I missing?

Reply:
No, the entire point of having unserialiazable containers is not to have a single chunk of 1MB to be transmitted over TCP. We are creating a `[][]byte` meaning an array of []byte chunks. This is the chunked array you talked about. Every chunk is already serialized (before arriving to the gossip) but sits somewhere else in memory. We don't serialize them together, we just take a bunch of pointers and we transmit every chunk separately.

This model can be changed to a Writer very easily. This is a great task for the gossip team, just change the interface of the transport from a `[][]byte` to a Writer. Pretty easy to do and does not require any architecture changes, it's just an implementation detail of gossip which is great.